lang/csharp/src/apache/test/Reflect/TestFromAvroProject.cs

```diff
@@ -0,0 +1,252 @@
+                    { ""name"" : ""myNull"", ""type"" : ""null"" },
+                    { ""name"" : ""myFixed"", ""type"" : ""MyFixed"" },
+                    { ""name"" : ""myA"", ""type"" : ""A"" },
+                    { ""name"" : ""myE"", ""type"" : ""MyEnum"" },
+                    { ""name"" : ""myArray"", ""type"" : { ""type"" : ""array"", ""items"" : ""bytes"" } },
+                    { ""name"" : ""myArray2"", ""type"" : { ""type"" : ""array"", ""items"" : { ""type"" : ""record"", ""name"" : ""newRec"", ""fields"" : [ { ""name"" : ""f1"", ""type"" : ""long""} ] } } },
+                    { ""name"" : ""myMap"", ""type"" : { ""type"" : ""map"", ""values"" : ""string"" } },
+                    { ""name"" : ""myMap2"", ""type"" : { ""type"" : ""map"", ""values"" : ""newRec"" } },
+                    { ""name"" : ""myObject"", ""type"" : [ ""MyEnum"", ""A"", ""null"" ] },
+                    { ""name"" : ""myArray3"", ""type"" : { ""type"" : ""array"", ""items"" : { ""type"" : ""array"", ""items"" : [ ""double"", ""string"", ""null"" ] } } }
+                ]
+            }
+        ]
+        }";
+
+        public Z SerializeDeserialize(Z z)
+        {
+            try
+            {
+                Protocol protocol = Protocol.Parse(_avroTestSchemaV1);
+                Schema schema = null;
+                foreach (var s in protocol.Types)
+                {
+                    if (s.Name == "Z")
+                    {
+                        schema = s;
+                    }
+                }
+
+                var avroWriter = new ReflectWriter<Z>(schema);
+                var avroReader = new ReflectReader<Z>(schema, schema);
+
+                byte[] serialized;
+
+                using (var stream = new MemoryStream(256))
+                {
+                    avroWriter.Write(z, new BinaryEncoder(stream));
+                    serialized = stream.ToArray();
+                }
+
+                Z deserialized = null;
+                using (var stream = new MemoryStream(serialized))
+                {
+                    deserialized = avroReader.Read(default(Z), new BinaryDecoder(stream));
+                }
+
+                return deserialized;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
+                throw ex;
+            }
+        }
+
+        [TestCase]
+        public void DefaultZ()
+        {
+            var z = new Z()
+            {
+                myBytes = new byte[10],
+                myString = "123",
+                myFixed = new byte[16],
+                myA = new A(),
+                myArray = new List<byte[]>(),
+                myArray2 = new List<newRec>(),
+                myMap = new Dictionary<string, string>(),
+                myMap2 = new Dictionary<string, newRec>(),
+                myArray3 = new List<List<object>>()
+            };
+
+            var zz = SerializeDeserialize(z);
```

Comment:
We should assert that all the properties of `zz` equal those of `z`.

Reply:
Done